File: leetgo/go/LCR 128.xuan-zhuan-shu-zu-de-zui-xiao-shu-zi-lcof/solution.go
// Created by k at 2024/05/13 15:23
// leetgo: dev
// https://leetcode.cn/problems/xuan-zhuan-shu-zu-de-zui-xiao-shu-zi-lcof/

package main

import (
	"bufio"
	"fmt"
	"os"

	. "github.com/j178/leetgo/testutils/go"
)

// @lc code=begin

func stockManagement(stock []int) (ans int) {
	ans = stock[0]
	for i := 0; i+1 < len(stock); i++ {
		if stock[i] > stock[i+1] {
			ans = stock[i+1]
		}
	}
	return
}

// @lc code=end

func main() {
	stdin := bufio.NewReader(os.Stdin)
	stock := Deserialize[[]int](ReadLine(stdin))
	ans := inventoryManagement(stock)

	fmt.Println("\noutput:", Serialize(ans))
}

File: leetgo/cpp/LCR 128.xuan-zhuan-shu-zu-de-zui-xiao-shu-zi-lcof/solution.cpp
// Created by k at 2024/05/13 15:32
// leetgo: dev
// https://leetcode.cn/problems/xuan-zhuan-shu-zu-de-zui-xiao-shu-zi-lcof/

#include <bits/stdc++.h>
#include "LC_IO.h"
using namespace std;

// @lc code=begin

class Solution {
  public:
    int stockManagement(vector<int> &stock)
    {
        for (int i = 0; i + 1 < stock.size(); i++) {
            if (stock[i] > stock[i + 1]) {
                return stock[i + 1];
            }
        }
        return stock[0];
    }
};

// @lc code=end

int main()
{
    ios_base::sync_with_stdio(false);
    stringstream out_stream;

    vector<int> stock;
    LeetCodeIO::scan(cin, stock);

    Solution *obj = new Solution();
    auto res      = obj->inventoryManagement(stock);
    LeetCodeIO::print(out_stream, res);
    cout << "\noutput: " << out_stream.rdbuf() << endl;

    delete obj;
    return 0;
}
